octobot_trading/exchanges/parser/orders_parser.py

```diff
@@ -0,0 +1,730 @@
+            f"({taker_or_maker or 'no taker_or_maker'}) which also failed to parse",
+        )
+
+    def _parse_exchange_order_type(self):
+        if octobot_order_type := self.formatted_record.get(
+            OrderCols.OCTOBOT_ORDER_TYPE.value
+        ):
+            try:
+                self.formatted_record[
+                    OrderCols.TYPE.value
+                ] = self.get_trade_order_type_from_map(octobot_order_type)
+            except (ValueError, KeyError):
+                self._log_missing(
+                    OrderCols.TYPE.value,
+                    f"failed to convert octobot order type ({octobot_order_type or 'no type'}) to order",
+                )
+        else:
+            self._log_missing(
+                OrderCols.TYPE.value,
+                f"not able to parse type without octobot order type ",
+            )
+
+    def get_trade_order_type_from_map(
+        self,
+        raw_order_type,
+    ):
+        return TradeOrderType(self.TRADE_ORDER_TYPE_MAP[raw_order_type]).value
+
+    def _parse_side(self, missing_side_value):
+        self._try_to_find_and_set(
+            OrderCols.SIDE.value,
+            self.SIDE_KEYS,
+            parse_method=self._found_side,
+            not_found_val=missing_side_value,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_SIDE,
+        )
+
+    def _found_side(self, raw_side):
+        return TradeOrderSide(self.TRADE_ORDER_SIDE_MAP[raw_side.lower()]).value
+
+    def _parse_price(self, missing_price_value):
+        self._try_to_find_and_set_decimal(
+            OrderCols.PRICE.value,
+            self.PRICE_KEYS,
+            not_found_val=missing_price_value,
+            enable_log=False if missing_price_value else True,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_PRICE,
+        )
+
+    def _parse_filled_price(self):
+        self._try_to_find_and_set_decimal(
+            OrderCols.FILLED_PRICE.value,
+            self.FILLED_PRICE_KEYS,
+            not_found_method=self.missing_filled_price,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_FILLED_PRICE,
+        )
+
+    def missing_filled_price(self, _):
+        if status := self.formatted_record.get(OrderCols.STATUS.value):
+            if (
+                status == OrderStatus.FILLED.value
+                or status == OrderStatus.CLOSED.value
+                or status == OrderStatus.PARTIALLY_FILLED.value
+            ) and (price := self.formatted_record.get(OrderCols.PRICE.value)):
+                return price
+            if (
+                status == OrderStatus.CANCELED.value
+                or status == OrderStatus.OPEN.value
+                or status == OrderStatus.PENDING_CANCEL.value
+                or status == OrderStatus.EXPIRED.value
+                or status == OrderStatus.PENDING_CREATION.value
+                or status == OrderStatus.REJECTED.value
+            ):
+                return 0  # to check - should we set it to None?
```

Comment:
price should be a decimal, I think it's safer to set it at 0 if we don't have it

Reply:
okay, I removed the comments, its always decimal